lib/PuppeteerSharp/Browser.cs

```diff
@@ -220,17 +222,15 @@ void TargetHandler(object sender, TargetChangedArgs e)
 
             try
             {
-                foreach (var target in Targets())
-                {
-                    if (predicate(target))
-                    {
-                        return target;
-                    }
-                }
-
                 TargetCreated += TargetHandler;
                 TargetChanged += TargetHandler;
 
+                existingTarget = Targets().FirstOrDefault(predicate);
+                if (existingTarget != null)
+                {
+                    return existingTarget;
+                }
```

Comment:
This [commit](https://github.com/hardkoded/puppeteer-sharp/pull/2051/commits/42cdc6550658dda8b807bdab89b72da36c60cb79) added enumeration of `Targets()` below attaching the event handlers - not sure if that was intended since the commit message is "Update chromium version".

[Puppeteer](https://github.com/puppeteer/puppeteer/blob/9abd48a062a4a30fb93d0b555f2fa03d3dc410f3/packages/puppeteer-core/src/common/Browser.ts#L499) only enumerates `targets` after attaching the event handlers.

Reply:
The first check for `Targets()` on line 207 is indeed unnecessary. I'll remove that.